### PR TITLE
401/Step 2: Define Spread

### LINKED ISSUE #401: Pooling interface: Step 2 - Select spread
Select the spread

Details: https://docs.google.com/document/d/1yNddJk6eW1NluEh5B8IL-UghDv9tWnl0ve17SgZBf1k/edit

image

Related to: https://github.com/gnosis/dex-react/issues/400

### PR DESCRIPTION
**Pull request by W3stside**
Closes #401 

1. Define spread in table form, responsive
2. uses map over object (just for you Leandro)

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr426--dexreact.review.gnosisdev.com

**Comment by alfetopito** [comment]
Also, the spread is reset to default when moving back from `Step 3`. I know this is coming as user feedback later on =P

File: src/components/PoolingWidget/DefineSpread.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      </tbody>
+    </CardTable>
+  </SpreadInformationWrapper>
+)
+
+const DefineSpread: React.FC<DefineSpreadProps> = ({ selectedTokensMap, defaultSpread = 0.2 }) => {
+  const [spread, setSpread] = useSafeState(defaultSpread)
+
+  const handleSpreadChange = ({ target: { value } }: React.ChangeEvent<HTMLInputElement>): void => {
+    if (+value < 0) return
```
[the remark this answers is not in the document]
Comment (W3stside) [reply src/components/PoolingWidget/DefineSpread.tsx]: yep, you're right

**Comment by W3stside** [comment]
unrelated, but does anyone know why this is showing me all the commits from the previous 400 branch i branched off of? I am 24 ahead of develop but seeing all the previous commits. Is it because I didn't rebase?

**Comment by W3stside** [comment]
> > unrelated, but does anyone know why this is showing me all the commits from the previous 400 branch i branched off of? I am 24 ahead of develop but seeing all the previous commits. Is it because I didn't rebase?
> 
> 400 is merged into develop so you expect 401 to be only commits different from develop + 400?
> But develop has one big squashed commit from 400, so all the small commits are not on develop and are detected as unique to 401.
> 
> You can try rebasing from develop, no idea if will work. Or dropping all commits up to [60e67f0](https://github.com/gnosis/dex-react/commit/60e67f08920ddb091ad7a0a2f3d50ffc87b22cff) in 401

sth for next time, thanks

**Comment by alfetopito** [comment]
> > unrelated, but does anyone know why this is showing me all the commits from the previous 400 branch i branched off of? I am 24 ahead of develop but seeing all the previous commits. Is it because I didn't rebase?
> 
> 400 is merged into develop so you expect 401 to be only commits different from develop + 400?
> But develop has one big squashed commit from 400, so all the small commits are not on develop and are detected as unique to 401.
> 
> You can try rebasing from develop, no idea if will work. Or dropping all commits up to [60e67f0](https://github.com/gnosis/dex-react/commit/60e67f08920ddb091ad7a0a2f3d50ffc87b22cff) in 401

I do that quite often. To help avoid issues during rebase, I first squash all commits form previous branch into a single commit, then rebase from origin/develop.

1. Get the commit id of the parent of the other branch
2. Rebase interactively: `rebase -i <commit id>`
3. Use `f` for all commits from the other branch, except the first
4. Rebase current branch against `develop`: `rebase origin/develop`
5. Should be nice and clean. Push with `-f` and change the tracking branch on PR to `develop`

**Comment by alfetopito** [comment]
I need this to continue with Step 3 (#402). Merged.
If there's any left over please just create a new PR :)

File: src/components/PoolingWidget/DefineSpread.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        // is a props and doesn't need !important
+        $webCSS={`
+        thead > tr { margin: 0 !important; };
+        tbody > tr { 
+          min-height: 3.5rem;
+        } 
+      `}
+        $responsiveCSS={`
+        tbody > tr { 
+          > td:first-child {
```
Comment (anxolin) [inline src/components/PoolingWidget/DefineSpread.tsx]: It feels a bit wrong this CSS injected in the component like this

**Pull request merged**